Source/WebCore/platform/win/WebCoreBundleWin.cpp

```diff
@@ -61,6 +66,92 @@ CFBundleRef webKitBundle()
     return bundle.get().get();
 }
 
-} // namespace WebCore
+String webKitBundlePath()
+{
+    URL bundleURL = adoptCF(CFBundleCopyBundleURL(webKitBundle())).get();
+    return bundleURL.fileSystemPath();
+}
 
-#endif
+String webKitBundlePath(StringView path)
+{
+    auto pathString = path.toStringWithoutCopying();
+    auto directory = FileSystem::parentPath(pathString);
+    auto fileName = FileSystem::pathFileName(pathString);
+
+    auto parts = fileName.split('.');
+    auto partsSize = parts.size();
+    if (partsSize == 2)
+        return webKitBundlePath(parts[0], parts[1], directory);
+
+    StringBuilder builder;
+    builder.append(parts[0]);
+    for (size_t i = 1; i < partsSize - 1; ++i) {
+        builder.append("."_s);
+        builder.append(parts[i]);
+    }
+
+    return webKitBundlePath(builder.toString(), parts[partsSize - 1], directory);
+}
+
+String webKitBundlePath(StringView name, StringView type, StringView directory)
+{
+    auto resourceURL = adoptCF(CFBundleCopyResourceURL(webKitBundle(), name.createCFString().get(), type.createCFString().get(), directory.createCFString().get()));
+    if (!resourceURL)
+        return { };
+
+    return adoptCF(CFURLCopyFileSystemPath(resourceURL.get(), kCFURLWindowsPathStyle)).get();
+}
+
+String webKitBundlePath(const Vector<StringView>& components)
+{
+    return webKitBundlePath(FileSystem::pathByAppendingComponents(""_s, components));
+}
+
+#else
+
+static String dllDirectory()
+{
+    WCHAR buffer[MAX_PATH];
+    DWORD length = ::GetModuleFileNameW(WebCore::instanceHandle(), buffer, MAX_PATH);
+    if (!length || (length == MAX_PATH && GetLastError() == ERROR_INSUFFICIENT_BUFFER))
+        return { };
+
+    String path(buffer, length);
+    return FileSystem::parentPath(path);
+}
+
+String webKitBundlePath()
+{
+    static NeverDestroyed<String> bundle = FileSystem::pathByAppendingComponent(dllDirectory(), "WebKit.resources"_s);
+    return bundle;
+}
+
+String webKitBundlePath(StringView path)
+{
+    auto resource = FileSystem::pathByAppendingComponent(webKitBundlePath(), path);
+    if (!FileSystem::fileExists(resource))
+        return emptyString();
```

Comment:
Do callers know how to deal with this empty string? How did we choose empty string rather than null string?